fatbin/create.go

```diff
@@ -21,25 +24,29 @@ type Writer struct {
 // attached to the binary. It relies on content sniffing (see Sniff)
 // to determine its offset.
```

Comment:
make it clear in the comments that the file must be opened with O_TRUNC, others the invalid file descriptor error is pretty confusing!

Reply:
Yeah, truncation is for that. But maybe it's not a useful feature?